### PR TITLE
Daemonset rolling update

### PR DESCRIPTION
**Pull request by Seb-Solon**
Hi,

This PR add a feature into the kubectl to update daemonset. 

It basically  do the following : 
- Create the new DS
- Delete the old one 
- Recreate pod associated to the selector.

Rolling update for daemonset implies to have the same selector, other it makes not sense to roll update (a simple create / delete should do the trick IMO). It also require a different name for the new daemon set (similar to RC tolling update)

I tried to follow example I found in kubernetes code to add this new feature. 

Feel free to give me some advice to improve code quality :smile:

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks. We'll take a look after 1.2.

Quick comments:
* Unlike rolling-update, creating a new DaemonSet should not be necessary. It should be possible to update the DaemonSet and delete the pods, because the pods otherwise normally won't get deleted.
* At some point, we plan to implement this server-side, as with Deployment. We're trying to reduce the amount of logic in kubectl. #12143

cc @mml @mikedanese @davidopp @madhusudancs @janetkuo @pwittrock

**Comment by Seb-Solon** [comment]
Hi there, 

Thanks for the reply. I've created a ticket to keep track of it and follow the guidelines for submitting a patch.

I wiill work on updating instead of creating / deleting no problem. Do you think I should start look at where to put this logique on the server side instead of the kubectl? 

Regards

File: pkg/kubectl/cmd/daemonsetrollingupdate.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+// RollingUpdateOptions is the start of the data required to perform the operation.  As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+//type RollingUpdateOptions struct {
+//	Filenames []string
+//}
+
+const (
+	daemonSetRollingUpdate_long = `Perform a rolling update of the given DaemonSet.
+
+Replaces the specified daemon set with a new daemon set by creating a daemon set with the same selector and delering the
```
Comment (fatih) [inline pkg/kubectl/cmd/daemonsetrollingupdate.go]: typo: `delering`

**Comment by bgrant0607** [comment]
I proposed a path forward on #22543

**Comment by bgrant0607** [comment]
The PR in its current form doesn't match what we want, so I'm going close it for now. Please reopen it or open a new one once there is an implementation that matches an agreed-upon approach. Let's discuss in the issue, though.

**Pull request closed without merging**